e2e/disk_setup.go

```diff
@@ -0,0 +1,104 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func diskSetup() func(*testing.T) {
+	return func(t *testing.T) {
```

Comment:
does this need to be a function inside another function ? For simplicity, I would prefer to keep it a single function if internal function is not needed here.